python/shotgun_menus/entity_field_menu.py

```diff
@@ -0,0 +1,311 @@
+
+        self._sg_entity_type = sg_entity_type
+
+        # default state
+        self._field_filter = None
+        self._checked_filter = None
+        self._disabled_filter = None
+
+        # prefix for fields if this menu represents an entity bubbled through another field
+        self._bubble_base = None
+
+        self._owns_task_manager = False
+        self._task_manager = bg_task_manager
+        if self._task_manager is None:
+            self._owns_task_manager = True
+
+            task_manager = sgtk.platform.import_framework("tk-framework-shotgunutils", "task_manager")
+            self._task_manager = task_manager.BackgroundTaskManager(
+                parent=self,
+                max_threads=1,
+                start_processing=True
+            )
+
+            shotgun_globals.register_bg_task_manager(self._task_manager)
+
+        # populate the menu the first time it is shown
+        self._initialized = False
+        self.aboutToShow.connect(self._on_about_to_show)
+
+    def set_field_filter(self, field_filter):
+        """
+        Set the callback used to filter which fields are shown by the menu
+
+        :param field_filter: Callback called for each entity field that returns True if the field
+            should be shown and False if it should not
+        :type field_filter: A method that takes a single field string as its only argument and
+            returns a boolean
+        """
+        self._field_filter = field_filter
+
+    def set_checked_filter(self, checked_filter):
+        """
+        Set the callback used to set which fields are checked.  By specifying a value other than
+        None, all the menu items will be checkable.
+
+        :param checked_filter: Callback called for each entity field that returns True if the field
+            should be checked and False if it should not
+        :type checked_filter: A method that takes a single field string as its only argument and
+            returns a boolean
+        """
+        self._checked_filter = checked_filter
+
+    def set_disabled_filter(self, disabled_filter):
+        """
+        Set the callback used to filter which fields are disabled
+
+        :param disabled_filter: Callback called for each entity field that returns True if the field
+            should be disabled and False if it should not
+        :type disabled_filter: A method that takes a single field string as its only argument and
+            returns a boolean
+        """
+        self._disabled_filter = disabled_filter
+
+    def add_label(self, title):
+        """
+        Add a label with the given title to the menu
+
+        :param title: The title of the sectional label
+        :type title: String
+        """
+        # build the label widget
+        label = QtGui.QLabel(self)
+        label.setStyleSheet("margin: 0.25em; color: gray;")
+        font_style = "font-variant: small-caps; font-weight: bold; font-size: large"
```

Comment:
Agreed.  Done and MUCH better.

<img width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/319506/13116589/50b7c704-d551-11e5-8dfd-ea71da9ea14c.png">